File: implementacion_java/src/utilidades/AlgoritmoTiempo.java
package utilidades;

public class AlgoritmoTiempo implements Comparable<AlgoritmoTiempo> {
    private final String nombre;
    private final double tiempo;

    public AlgoritmoTiempo(String nombre, double tiempo) {
        this.nombre = nombre;
        this.tiempo = tiempo;
    }

    public String getNombre() {
        return nombre;
    }

    public double getTiempo() {
        return tiempo;
    }

    @Override
    public int compareTo(AlgoritmoTiempo otro) {
        return Double.compare(otro.tiempo, this.tiempo);
    }
}

File: implementacion_kotlin/src/utilidades/AlgoritmoTiempo.kt
package utilidades

class AlgoritmoTiempo(val nombre: String, val tiempo: Double) : Comparable<AlgoritmoTiempo> {
    override fun compareTo(otro: AlgoritmoTiempo): Int {
        return java.lang.Double.compare(otro.tiempo, this.tiempo)
    }
}